### PR TITLE
Universal icon usage and allow for license check.

### PR DESCRIPTION
**Pull request by ntoll**
This PR introduces three updates:

1. If an icon is specified it will be used in the install/uninstall process and as the icon for the installer itself.
2. If a `license` path is given in the `Application` section of the configuration file an additional step will take place before installation to check the user's agreement to abide by the displayed license. If the license is not given, the extra step is omitted (the default behaviour).
3. Documentation updates to reflect the above two changes.

### CODE REVIEW

File: doc/cfgfile.rst [hunk 1]
```diff
@@ -80,6 +81,13 @@ Application section
    If you use the Python API, this parameter can also be passed as a file-like
    object, such as :class:`io.StringIO`.
 
+.. describe:: license (optional)
```
Comment (takluyver) [inline doc/cfgfile.rst]: For the sake of explicitness, I like your earlier suggestion to call this `license_file` - that makes it clear that it's a path, not just a name like `GPL`.

File: nsist/__init__.py [hunk 2]
```diff
@@ -282,7 +283,7 @@ def prepare_shortcuts(self):
         copy to the build directory. Prepare target and parameters for these
         shortcuts.
 
-        Also copies shortcut icons
+        Also copies shortcut icons, and, if specified, the license file.
```
Comment (takluyver) [inline nsist/__init__.py]: It doesn't make sense to me to add this to `prepare_shortcuts()`, but I don't think any of the other methods are right either. Let's add a new `copy_license()` method for this.

**Comment by takluyver** [comment]
Meant to point to the release notes file: https://github.com/takluyver/pynsist/blob/master/doc/releasenotes.rst

**Comment by ntoll** [comment]
@takluyver OK... all done. I've bumped it to version 2.2 since 2.1 is already out. Happy to change anything else if need be.

**Comment by ntoll** [comment]
@takluyver hey hey... any reasons this hasn't been merged yet..? Would like to use these updates in my automated build in appveyor (i.e. I can `pip install pynsist` with these changes). :-)

**Comment by takluyver** [comment]
No, I just got busy with other things and forgot about it. Pressing buttons now...

**Comment by takluyver** [comment]
OK, 2.1 is now out. :-)

**Pull request merged**